opennlp-tools/src/main/java/opennlp/tools/stemmer/snowball/SnowballStemmer.java

```diff
@@ -22,23 +22,42 @@
 public class SnowballStemmer implements Stemmer {
 
   public enum ALGORITHM {
-    DANISH,
-    DUTCH,
-    ENGLISH,
-    FINNISH,
-    FRENCH,
-    GERMAN,
-    HUNGARIAN,
-    IRISH,
-    ITALIAN,
-    NORWEGIAN,
-    PORTER,
-    PORTUGUESE,
-    ROMANIAN,
-    RUSSIAN,
-    SPANISH,
-    SWEDISH,
-    TURKISH
+    DANISH("dan"),
+    DUTCH("nld"),
+    ENGLISH("eng"),
+    FINNISH("fin"),
+    FRENCH("fra"),
+    GERMAN("deu"),
+    HUNGARIAN("hun"),
+    IRISH("gle"),
+    ITALIAN("ita"),
+    NORWEGIAN("nor"),
+    PORTER("porter"),
+    PORTUGUESE("por"),
+    ROMANIAN("ron"),
+    RUSSIAN("rus"),
+    SPANISH("spa"),
+    SWEDISH("swe"),
+    TURKISH("tur");
+
+    private String languageCode;
+
+    private ALGORITHM(String languageCode) {
+      this.languageCode = languageCode;
+    }
+
+    public String getLanguageCode() {
+      return languageCode;
+    }
+
+    public static ALGORITHM getByLanguageCode(String languageCode) {
```

Comment:
It has to say by ISO-639-3 language code. Otherwise people might try two letter codes. Maybe it would be better to pass in here LanguageCode as well.